azure-iot-provisioning-devicesdk/azure/iot/provisioning/devicesdk/registration_client_factory.py

```diff
@@ -0,0 +1,39 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+"""
+This module provides factory methods for creating clients which would
+communicate with Device Provisioning Service.
+"""
+from .security.sk_security_client import SymmetricKeySecurityClient
+from .transport.mqtt_transport import MQTTTransport
+from .sk_registration_client import SymmetricKeyRegistrationClient
+
+
+def create_from_security_client(provisioning_host, security_client, transport_choice):
```

Comment:
>provisioning_host [](start = 32, length = 17)

this will also need to accept an id_scope (unless that concept is gone or the user passes it in some other way that I haven't come across yet).

Reply:
id scope is inside the security client

---
In reply to: [271430125](https://github.com/Azure/azure-iot-sdk-python-preview/pull/59#discussion_r271430125) [](ancestors = 271430125)